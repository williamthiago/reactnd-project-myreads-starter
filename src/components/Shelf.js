import React from 'react'

import Book from './Book'

const Shelf = ({title, books}) => (
  <div className="bookshelf">
    <h2 className="bookshelf-title">{title}</h2>
    <div className="bookshelf-books">
      <ol className="books-grid">
        {books.map(book => (
          <li key={book.id}>
            <Book book={book} />
          </li>
        ))}
      </ol>
    </div>
  </div>
)

export default Shelf