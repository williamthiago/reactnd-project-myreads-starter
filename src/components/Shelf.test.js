import React from 'react'
import { shallow } from 'enzyme'

import Shelf from './Shelf'

it('Renders Shelf without books', () => {
  const books = []
  const onChangeShelf = () => {}
  expect(shallow(<Shelf books={books} title="Shelf Test" onChangeShelf={onChangeShelf} />)).toMatchSnapshot()
})