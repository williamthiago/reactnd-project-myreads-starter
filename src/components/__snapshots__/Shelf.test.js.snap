// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Shelf without books 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Shelf Test
  </h2>
  <div
    className="bookshelf-books"
  >
    <div
      className="bookshelf-empty"
    >
      <img
        alt=""
        src="no-books.png"
      />
      <p>
        Oops... There are no books!
      </p>
    </div>
  </div>
</div>
`;
